### PR TITLE
[FIX] event{._sms}: irrelevant template data display

### PR DESCRIPTION
**Pull request by joanna350**
1. Install Events
2. Click on the app
- click on any event
- Communication tab
- select [Social Post] (one without demo data) on Send column
- click on Template section to get the dropdown

Issue: what you see are not relevant to the Send section
Resolve by: Give some guidance to the user
Linked w: [PR#38684](https://github.com/odoo/enterprise/pull/38684)
Impacted version: 15.0 - master
opw-3103199

### CODE REVIEW

**Comment by nle-odoo** [comment]
> @hale-odoo @nle-odoo this is getting ridiculous ... we should not have 4 PRs to fix an access to ir.model. How do you end up fixing the fix of the fix ? Don't you test before r+ ?

This is separate ticket (opw-3103199) that was reported 3 months ago and happen before and after the merged PR that fix access error for non-admin.

The issue is that if there is no record to show (for sms / mail / social) you can select other types of records (eg: a mail.template for a "SMS" type) with the wrong notification_type.

**Comment by tde-banana-odoo** [comment]
Ok sorry, somehow I ended up on the old bugfix trying to get up in the fix-and-issue chain. Maybe I did something wrong then, hence removing the comment.

**Comment by nle-odoo** [comment]
> Ok sorry, somehow I ended up on the old bugfix trying to get up in the fix-and-issue chain. Maybe I did something wrong then, hence removing the comment.

You are correct, in this ticket originally there was the issue of this PR. And the other PR was communicated but did not change the original reported issue.

File: addons/event/models/event_mail.py [hunk 1]
```diff
@@ -84,8 +85,12 @@ def _selection_template_model(self):
...
         mail_model = self.env['mail.template']
         if self.notification_type == 'mail':
-            record = mail_model.search([('model', '=', 'event.registration')], limit=1)
-            self.template_ref = "{},{}".format('mail.template', record.id) if record else False
+            template = mail_model.search([('model', '=', 'event.registration')], limit=1)
+            if template:
+                self.template_ref = "{},{}".format('mail.template', template.id)
+            else:
+                raise UserError(
+                    _("There is no template data relevant to the notification type."))
```
Comment (nle-odoo) [inline addons/event/models/event_mail.py:93]: translation should be added in addons/event/i18n/event.pot

**Review by nle-odoo — APPROVED** [review]
seems good to me

as tde said, there seems to be many issue for this part, so adding test would be good

for this part, we could add a test:

- in event_sms (since you need two notification types) setting sms without sms.template trigger the error and if we save it after the error it saves "EMAIL" as notification type
- in event_sms (since you need two notification types) setting email without sms.template trigger the error and if we save it after the error it saves "SMS" as notification type

**Comment by thiryr** [comment]
This approach is a bit problematic in that people would have normally been able to create a template from there. And now they'll just get an error and have to figure out some other way.

I believe the core of the issue is actually the recent fix we discussed and approved recently here fe80a2676a1c5f964c20740ae8097929564a1858

By removing `model_field` from the options we made the reference field fall back on the currently-selected record to determine which model the associated m2o field should use. As a result if there is no change in the template value, the field keeps the last model it could find.

-----------------------

I was about to create a fix for this where we would simply switch to using the model picker of the reference field, just removing 'hide_model' from the reference field option. And set the model as a computed field. It seems doable on stable without upgrade script. I'm not sure why we didn't do that to start with to be honest.

**Comment by nle-odoo** [comment]
> Opw tagged here is wrong, I raised the PR for it separately.

I closed the new branch (https://github.com/odoo/odoo/pull/117090/files) and reopened this one, we should not create a new PR and lose PR comment history just to change the name of the branch.

> I was about to create a fix for this where we would simply switch to using the model picker of the reference field, just removing 'hide_model' from the reference field option. And set the model as a computed field. It seems doable on stable without upgrade script. I'm not sure why we didn't do that to start with to be honest.

Seems like good solution if it works :+1: 

So adding a new compute @api.depends('notification_type') that compute `template_ref` and `template_model_id`.

I think in the original PR we thought about that, but there is still the information of `template_model_id.name` that was missing (and so would need to be gotten by "a route / a method / a non-stored computed field").

**Comment by thiryr** [comment]
Making a POC should be more clear:

https://github.com/odoo/odoo/pull/117153

We already have the 'models' in the front-end thanks to the selection used by the reference field. Since users can only select from those models we can safely use sudo (implicitly in the stored compute) to get the model for wherever it is needed.

And since the reference field is required we know we can always rely on it (in the POC I set a value in case it doesn't exist because in effect changing the model triggers a compute for the front-end)

I think this should be ok without update script (but would require a -u, obviously)

**Comment by thiryr** [comment]
Hey, as discussed with hale we'll be taking this over as a task since it's apparently not directly linked to the ticket mentioned in the commit message. It could be worth mentioning on the ticket in question, leaving it up to you.

Moving the discussion to the POC PR linked above since it seems to be the direction we're heading and I'd rather get some validation from our PO and the rest of the team for a relatively important change on stable.

Thanks for your inputs, and thanks @hale-odoo for investigating.
New task is 3256524 and I tagged @hale-odoo on there.

Feel free to close the PR again unless there's something else :)

**Pull request closed without merging**